backend/routes/offline-payment.js

```diff
@@ -79,11 +81,43 @@ module.exports = function (router) {
         return res.json({ success: false, message: 'Missing order data' })
       }
 
+      const paymentCode = randomstring.generate()
+
       req.body.data = encryptedData
       req.amount = 0
-      req.paymentCode = randomstring.generate()
+      req.paymentCode = paymentCode
+      req.paymentType = OrderPaymentTypes.Offline
+
       next()
     },
     makeOffer
   )
+
+  router.put(
+    '/offline-payments/payment-state',
+    authSellerAndShop,
+    async (req, res) => {
+      const { paymentCode, state } = req.body
+
+      const order = await Order.findOne({
+        where: {
+          shopId: req.shop.id,
+          paymentCode,
+          paymentType: OrderPaymentTypes.Offline
+        }
+      })
+
+      if (!order) {
+        return res.status(200).send({
+          reason: 'Invalid payment code'
+        })
+      }
+
+      await order.update({
+        paymentStatus: state
```

Comment:
If we want to be paranoid, we could add some checks on the new status value. To prevent invalid transitions. For example an order that is in payment status "Paid" should never be moved back to "Pending". Leaving a TODO is probably fine for now.